ios/RCTTWVideoModule.m

```diff
@@ -298,7 +325,32 @@ - (bool)_setLocalVideoEnabled:(bool)enabled cameraType:(NSString *)cameraType {
                 }
             }
         }];
-  }
+    }
+}
+
+RCT_EXPORT_METHOD(prepareToRebuildLocalVideoTrack) {
+    NSLog(@"[RNTwilioVideo] Preparing to rebuild local video track");
+    TVIVideoView *currentRenderer;
+
+    if (self.localVideoTrack != nil) {
+        [self.localVideoTrack setEnabled:false];
+        for (TVIVideoView *renderer in self.localVideoTrack.renderers) {
```

Comment:
❓ Why is the teardown outside of the for-loop if we're trying to tear down the renderers from the `localVideoTrack`? Is this necessary because we want to prevent zombie renderers when we set the property to `nil`?

Reply:
Its not a quite a teardown, but a transfer of the `renderer` to the new video track we are about to create.  For the other part of your question, many languages prevent you from editing a list while currently iterating it, and I wasn't sure if objective-c would work okay, error out, or have other random weirdness when doing this.  So I kept it safe and just edited the list after completing the iteration.

Reply:
This will get interesting when we move to capture frame changes.

In my first pass at adding capture frame functionality, I added a custom renderer for ios to the video track, so the video track would in that case have 2 renderers attached and this logic would break

https://github.com/blackuy/react-native-twilio-video-webrtc/pull/709/files#diff-095ec603ace0f7b8e01f138dba01a0c89488855a73e16f2b4fc313a7dd79e55e

We can cross that bridge when we get there but just pointing it out from seeing this